salt/cloud/clouds/ec2.py

```diff
@@ -4374,6 +4375,40 @@ def delete_keypair(kwargs=None, call=None):
                      sigver='4')
     return data
 
+def describe_keypairs(call=None):
```

Comment:
@kvnaveen Since this PR contains both a bug fix and a feature addition, would you mind splitting this up into 2 pull requests? The bug fix portion for `list_nodes_full` should be against `2017.7` as you have above. But these two new functions should be added as new features to the `develop` branch.

Please let us know if you have any questions. :)